DakeAndDevileCorps/Screens/MapHome/Component/SearchBarView.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  SearchBarView.swift
+//  DakeAndDevileCorps
+//
+//  Created by Kim Sujin on 2022/07/20.
+//
+
+import UIKit
+
+@objc protocol SearchBarDelegate {
+    @objc optional func didReturnKeyInput()
+    @objc optional func didBeginEditing()
+    @objc optional func touchUpInsideLeftButton()
+}
+
+class SearchBarView: UIView {
+    
+    weak var delegate: SearchBarDelegate?
+    var leftItemMode: LeftItemMode = .imageMode {
+        didSet {
+            switch leftItemMode {
+            case .imageMode:
+                symbolImageView.isHidden = false
+                leftButton.isHidden = true
+            case .buttonMode:
+                symbolImageView.isHidden = true
+                leftButton.isHidden = false
+            }
+        }
+    }
+    var image: UIImage? = UIImage(systemName: "magnifyingglass") {
+        didSet {
+            symbolImageView.image = image
+            leftButton.setImage(image, for: .normal)
+        }
+    }
+    
+    private lazy var symbolImageView: UIImageView = {
+        $0.translatesAutoresizingMaskIntoConstraints = false
+        $0.image = image
+        $0.contentMode = .scaleAspectFit
+        if leftItemMode == .imageMode {
+            $0.isHidden = false
+        } else { $0.isHidden = true }
+        $0.tintColor = .black
+        return $0
+    }(UIImageView())
```

Comment:
p5;

```suggestion
    private lazy var symbolImageView: UIImageView = {
        let imageView = UIImageView()
        imageView.translatesAutoresizingMaskIntoConstraints = false
        imageView.image = image
        imageView.contentMode = .scaleAspectFit
        imageView.isHidden = !(leftItemMode == .imageMode)
        imageView.tintColor = .black
        return imageView
    }()
```
혹시 하단에 넣지 않고 이렇게 쓰는 건 어떨까요? 저는 이 방식으로 사용하는데, 어차피 return를 할 거면 위에서 선언해서 return하는게 더 좋지 않을까 싶어서요!

Reply:
저는 만들고자 하는 뷰를 먼저 만들고, 해당 뷰를 필요에 맞게 세팅하는 것 같아서 클로저의 파라미터로 넣는 방식을 좋아했는데,
듣고보니 위에서 선언하고 밑에서 return하는 방식으로 코드를 짜면 조금 더 코드 흐름이 구조적으로 느껴질 것 같네요.
반영하겠습니다.

Reply:
@skycat0212 
뜬금없는 말이지만, 저렇게 사용할 수 있다는걸 처음 알았습니다. 저 방식도 신기하네요...